### PR TITLE
DOC: Improve documentation of scipy.signal.signaltools.wiener

### LINKED ISSUE #11415: Where is the Wiener Filter derived from?
I am using the Wiener Filter to do speech enhancement (1D data) and while I see the [source code](https://github.com/scipy/scipy/blob/v0.14.0/scipy/signal/signaltools.py#L441), I'm unsure where it was derived from. Is there an academic or authoritative source that it came from?

### PR DESCRIPTION
**Pull request by grapheo12**
#### Reference issue

Closes #11415 

#### What does this implement/fix?

- Docstring of the wiener function has been updated to show the source and equivalent Matlab/Octave function

### CODE REVIEW

**Comment by grapheo12** [comment]
Am I proceeding in the right direction? Or do I need to add something more to the docstring?
Should I proceed with the documentation? @pv @ilayn

File: scipy/signal/signaltools.py [hunk 1]
```diff
@@ -1412,6 +1412,10 @@ def wiener(im, mysize=None, noise=None):
     """
     Perform a Wiener filter on an N-dimensional array.
 
+    It is similar to wiener2 in Matlab/Octave.
+    Refer to Lim, Jae S., Two-Dimensional Signal and Image Processing,
+    Englewood Cliffs, NJ, Prentice Hall, 1990, p. 548.
+
     Apply a Wiener filter to the N-dimensional array `im`.
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: Put the explanation under this line.

File: scipy/signal/signaltools.py [hunk 2]
```diff
@@ -1412,6 +1412,10 @@ def wiener(im, mysize=None, noise=None):
     """
     Perform a Wiener filter on an N-dimensional array.
 
+    It is similar to wiener2 in Matlab/Octave.
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: Could make this a bit more expressive via putting it in the Notes section

```
.... some existing text ...

    Returns
    -------
    out : ndarray
        Wiener filtered result with the same shape as `im`.

    Examples
    --------
    ... some examples would be really appreciated see other
        functions in the same file for examples

    Notes
    -----
    This implementation is similar to Matlab/Octave's `wiener2` function. 
    For more details see [1]_

    References
    ------------
    .. [1] Lim, Jae S., Two-Dimensional Signal and Image Processing,
           Englewood Cliffs, NJ, Prentice Hall, 1990, p. 548.

    """
```

**Review by ilayn — CHANGES_REQUESTED** [review]
Thanks @grapheo12 for taking the time to submit a PR. If you don't mind could you please add some basic examples too while we are at it? I've placed some boilerplate structure in the comments.

**Comment by grapheo12** [comment]
@ilayn Please review.

**Comment by grapheo12** [comment]
Please suggest what should I do:
I intend to show 2 images side by side using `plt.subplots`. It is working fine in a notebook or from shell. But when the docs are compiled by Sphinx, only one image is shown.

File: scipy/signal/signaltools.py [hunk 3]
```diff
@@ -1439,10 +1439,14 @@ def wiener(im, mysize=None, noise=None):
...
     >>> import matplotlib.pyplot as plt
     >>> from pylab import *
-    >>> x = np.random.random((40,40))         #Create a random image
-    >>> plot1 = plt.imshow(x)                 #Plot the image
-    >>> #Show image filtered with 5px by 5px window
-    >>> plot2 = plt.imshow(wiener(x, (5, 5)))
+    >>> img = np.random.random((40,40))    #Create a random image
+    >>> #Filter image with 5px by 5px window
+    >>> filtered_img = wiener(img, (5, 5))
+    >>> plt.figure()
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: This creates a figure that you don't use. In the line below you get `f` as the figure anyways so creates a dummy output.

File: scipy/signal/signaltools.py [hunk 4]
```diff
@@ -1439,10 +1439,14 @@ def wiener(im, mysize=None, noise=None):
     >>> import numpy as np
     >>> import matplotlib.pyplot as plt
     >>> from pylab import *
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: We don't need this line anymore right? That's unnecessary import of the whole namespace

File: scipy/signal/signaltools.py [hunk 5]
```diff
@@ -1439,10 +1439,14 @@ def wiener(im, mysize=None, noise=None):
...
-    >>> x = np.random.random((40,40))         #Create a random image
-    >>> plot1 = plt.imshow(x)                 #Plot the image
-    >>> #Show image filtered with 5px by 5px window
-    >>> plot2 = plt.imshow(wiener(x, (5, 5)))
+    >>> img = np.random.random((40,40))    #Create a random image
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: We can be a bit more economic here

```python
>>> img = np.random.random((40,40))  # Create a random image
>>> filtered_img = wiener(img, (5, 5))  # Filter the image
....
```

File: scipy/signal/signaltools.py [hunk 6]
```diff
@@ -1439,10 +1439,14 @@ def wiener(im, mysize=None, noise=None):
...
+    >>> #Filter image with 5px by 5px window
+    >>> filtered_img = wiener(img, (5, 5))
+    >>> plt.figure()
+    >>> f, arr = plt.subplots(1, 2)
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: You can also add figure size to control the whitespace if the default is too big. See the example here https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.fourier_gaussian.html

**Comment by grapheo12** [comment]
I have made the changes as requested. Thanks for being patient with me.

**Review by ilayn — APPROVED** [review]

**Comment by ilayn** [comment]
My pleasure, thank you for your efforts @grapheo12 !

**Comment by mdhaber** [comment]
Thanks @grapheo12! If you'd like your name added to [THANKS.txt](https://github.com/scipy/scipy/blob/master/THANKS.txt), please make an entry [here](https://github.com/scipy/scipy/wiki/*THANKS.txt-additions-modifications-for-1.5.0*).

**Pull request merged**